### PR TITLE
removed gogo-proto extensions from *.proto

### PR DESCRIPTION
**Pull request by squat**
Signed-off-by: rahulii <r.sawra@gmail.com>

### CODE REVIEW

**Comment by rahulii** [comment]
1. Removed gogo-specific extensions from rpc.proto
2. make proto

**Comment by rahulii** [comment]
While using gogoproto , the InfoResponse was as follows:
```cmd
type InfoResponse struct {
	// Deprecated. Use label_sets instead.
	Labels    []github_com_thanos_io_thanos_pkg_store_labelpb.ZLabel `protobuf:"bytes,1,rep,name=labels,proto3,customtype=github.com/thanos-io/thanos/pkg/store/labelpb.ZLabel" json:"labels"`
	MinTime   int64                                                  `protobuf:"varint,2,opt,name=min_time,json=minTime,proto3" json:"min_time,omitempty"`
	MaxTime   int64                                                  `protobuf:"varint,3,opt,name=max_time,json=maxTime,proto3" json:"max_time,omitempty"`
	StoreType StoreType                                              `protobuf:"varint,4,opt,name=storeType,proto3,enum=thanos.StoreType" json:"storeType,omitempty"`
	// label_sets is an unsorted list of `ZLabelSet`s.
	LabelSets []labelpb.ZLabelSet `protobuf:"bytes,5,rep,name=label_sets,json=labelSets,proto3" json:"label_sets"`
```

after removing gogoproto:
```cmd
type InfoResponse struct {
	// Deprecated. Use label_sets instead.
	Labels    []*labelpb.Label `protobuf:"bytes,1,rep,name=labels,proto3" json:"labels,omitempty"`
	MinTime   int64            `protobuf:"varint,2,opt,name=min_time,json=minTime,proto3" json:"min_time,omitempty"`
	MaxTime   int64            `protobuf:"varint,3,opt,name=max_time,json=maxTime,proto3" json:"max_time,omitempty"`
	StoreType StoreType        `protobuf:"varint,4,opt,name=storeType,proto3,enum=thanos.StoreType" json:"storeType,omitempty"`
	// label_sets is an unsorted list of `ZLabelSet`s.
	LabelSets            []*labelpb.ZLabelSet `protobuf:"bytes,5,rep,name=label_sets,json=labelSets,proto3" json:"label_sets,omitempty"`
	XXX_NoUnkeyedLiteral struct{}             `json:"-"`
	XXX_unrecognized     []byte               `json:"-"`
	XXX_sizecache        int32                `json:"-"`
}
```
https://github.com/thanos-io/thanos/blob/main/pkg/store/prometheus.go#L112

in the above lines of code , should I replace 
`res := &storepb.InfoResponse{
		Labels:    make([]labelpb.ZLabel, 0, len(lset)),`

with 
`res := &storepb.InfoResponse{
		Labels:    make([]*labelpb.Label, 0, len(lset)),`

https://github.com/thanos-io/thanos/blob/main/pkg/store/prometheus.go#L118

but I am not sure , how should we handle this line where we are appending ZLabelsFromPromLabels()?

cc : @GiedriusS @squat

File: pkg/rules/manager.go [hunk 1]
```diff
@@ -41,13 +41,15 @@ type Group struct {
 }
 
 func (g Group) toProto() *rulespb.RuleGroup {
+	lastEvaluation,_ := types.TimestampProto(g.GetLastEvaluation())
```
Comment (GiedriusS) [inline pkg/rules/manager.go]: What about .UTC() here? (: Not sure if it makes a difference, reviewing on my phone

Comment (rahulii) [reply pkg/rules/manager.go]: noted. will add a UTC()

File: pkg/rules/rulespb/custom.go [hunk 2]
```diff
@@ -54,11 +53,11 @@ func NewRecordingRule(r *RecordingRule) *Rule {
 //
 // Note: This method assumes r1 and r2 are logically equal as per Rule#Compare.
 func (r1 *RecordingRule) Compare(r2 *RecordingRule) int {
-	if r1.LastEvaluation.Before(r2.LastEvaluation) {
+	if r1.LastEvaluation.Compare(r2.LastEvaluation) < 0 {
```
Comment (rahulii) [inline pkg/rules/rulespb/custom.go]: @GiedriusS @squat , wdyt about this `Compare()` in replacement for go time.Time's `Before() and After()`???

File: pkg/targets/targetspb/custom.go [hunk 3]
```diff
@@ -7,10 +7,18 @@ import (
...
 
+	"time"
+
+	"github.com/gogo/protobuf/types"
 	"github.com/pkg/errors"
 	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/thanos-io/thanos/pkg/rules/rulespb"
 	"github.com/thanos-io/thanos/pkg/store/labelpb"
 )
+func timeToProtoTimestamp(t time.Time) *types.Timestamp{
```
Comment (GiedriusS) [inline pkg/targets/targetspb/custom.go]: I think this is unused? :thinking:

File: .vscode/configurationCache.log [hunk 4]
```diff
@@ -0,0 +1 @@
...523-657ba532e42f","/home/infracloud/go/bin/prometheus-v1.8.2-0.20210331101223-3cafc58827d1","/home/infracloud/go/bin/prometheus-v2.4.3+incompatible","/home/infracloud/go/bin/promtool-v1.8.2-0.20200522113006-f4dd45609a05","/home/infracloud/go/bin/promu-v0.5.0","/home/infracloud/go/bin/protoc-3.4.0","/home/infracloud/go/bin/protoc-gen-gogofast-v1.3.2","/home/infracloud/go/bin/shfmt-v3.1.2","all","assets","build","check-docs","check-examples","check-git","crossbuild","deps","docker","docker-multi-stage","docker-push","docs","example-rules-lint","examples","examples-clean","examples/alerts/alerts.yaml","examples/alerts/rules.yaml","examples/dashboards","examples/tmp","format","go-format","go-lint","help","install-deps","jsonnet-format","jsonnet-lint","jsonnet-vendor","lint","mixin/runbook.md","pkg/ui/react-app/node_modules","pkg/ui/static/react","proto","quickstart","react-app-lint","react-app-lint-fix","react-app-start","react-app-test","shell-format","shell-lint","tarballs-release","test","test-e2e","test-e2e-local","test-local","web","web-pre-process","web-serve"],"launchTargets":[],"customConfigurationProvider":{"workspaceBrowse":{"browsePath":[],"compilerArgs":[]},"fileIndex":[]}}
```
Comment (GiedriusS) [inline .vscode/configurationCache.log]: What are these `.vscode/` files? :joy:

**Pull request merged**